### DISCUSSION TITLE
How to set up 301 redirects for canonical domain and trailing slash?

### DISCUSSION THREAD

**Discussion by VazgenZadayan** [vercel/next.js#73171] [Help]
### Summary

Hello everyone,

I need help configuring 301 redirects for my website my.company. Here's what I'm trying to achieve:

Redirect all the following variations to https://my.company:

www.my.company
http://my.company
http://www.my.company
https://www.my.company
Ensure that if a URL is missing a trailing slash (/), it should redirect to the same URL with the trailing slash.

For example:

https://my.company/page → https://my.company/page/

I'm looking for the best way to implement this.

Thank you!

**Comment by icyJoseph** [comment]
- https://nextjs.org/docs/app/api-reference/next-config-js/trailingSlash

> By default Next.js will redirect URLs with trailing slashes to their counterpart without a trailing slash. For example /about/ will redirect to /about. **You can configure this behavior to act the opposite way, where URLs without trailing slashes are redirected to their counterparts with trailing slashes.**

**Comment by denis23x** [comment]
Hi @VazgenZadayan 

For redirect to `www` and vice versa it is right to use domain settings, not the application.

Try to take a look there, probably you can find the solution.

**Reply by icyJoseph** [reply]
I mean, IIRC, technically you could do it in Middleware too, it's hacky though, and bug prone, like forgetting to forward all parts of the URL and handling dev Mode, etc
